docarray/array/any_array.py

```diff
@@ -51,8 +51,13 @@ def __repr__(self):
     @classmethod
     def __class_getitem__(cls, item: Union[Type[BaseDocWithoutId], TypeVar, str]):
         if not isinstance(item, type):
-            return Generic.__class_getitem__.__func__(cls, item)  # type: ignore
-            # this do nothing that checking that item is valid type var or str
+            try:
+                return Generic.__class_getitem__.__func__(cls, item)  # type: ignore
+                # this do nothing that checking that item is valid type var or str
+                # Keep the approach in #1147 to be compatible with lower versions of Python.
+            except AttributeError:
+                from types import GenericAlias
+                return GenericAlias(cls, item)  # type: ignore
```

Comment:
```suggestion
            python_minor_version = sys.version_info.minor
            if python_minor_version < 12:
                return Generic.__class_getitem__.__func__(cls, item)  # type: ignore
                # this do nothing that checking that item is valid type var or str
                # Keep the approach in #1147 to be compatible with lower versions of Python.
            else:
                 from types import GenericAlias
                 return GenericAlias(cls, item)  # type: ignore
```

Reply:
> Maybe ```sys.version_info < (3, 12)``` is better.

Reply:
yes true

Reply:
let's keep the computation of the boolean on the top of the file.